card/mod/03_machines/lib/javascript/wagn_mod.js.coffee

```diff
@@ -304,6 +303,27 @@ $(window).ready ->
        $(this).find('.glyphicon').addClass("glyphicon-ok").removeClass("glyphicon-remove")
        $(this).addClass("btn-success").removeClass("btn-danger"))
 
+  $('body').on 'click', '.follow-toggle', ->
+    anchor = $(this)
+    url  = wagn.rootPath + '/update/' + anchor.data('rule_name') + '.json'
+    $.ajax url, {
+      type : 'POST'
+      dataType : 'json'
+      data : {
+        'card[content]' : '[[' + anchor.data('follow').content + ']]'
+        'success[view]' : 'follow_status'
+        'success[id]'   : anchor.data('card_key')
+      }
+      success : (data)->
+        tags = anchor.closest('.card-menu').find('.follow-toggle')
+        tags.find('.follow-verb').html data.verb
+        tags.attr 'title', data.title
+        tags.attr 'class', data.class
+        tags.data 'follow', data
+    }
+    event.preventDefault() # Prevent link from following its href
+    
+
```

Comment:
this does all the work of making the request, interpreting the json results, and updating the links